rust/kernel/delay.rs

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Delay functions for operations like sleeping.
+//!
+//! C header: [`include/linux/delay.h`](../../../../include/linux/delay.h)
+
+use crate::{bindings, types::Milliseconds};
+use core::cmp::min;
+
+/// Sleeps safely even with waitqueue interruptions.
+///
+/// This function forwards the call to the C side `msleep` function.
+pub fn msleep(duration: Milliseconds) {
+    // If `duration` is chosen, `duration` is less than or equal to `c_uint::MAX`
```

Comment:
What this comment is trying to communicate is that the `min()` and then `as` conversion will not cause integer truncation but instead is used for saturation. If it's confusing, I can make it clearer that it's saturating instead of truncating. That might be a less wordy and easier to understand way to communicate it.